src/client/editor/objects/Model.js

```diff
@@ -6,21 +6,39 @@ export default class Model extends THREE.Object3D {
     super();
     this.type = "Model";
 
-    this.model = null;
+    this._src = null;
     this.animations = [];
     this.clipActions = [];
-
     this._mixer = new THREE.AnimationMixer(this);
   }
 
-  setModel(object, animations) {
-    this.model = object;
+  get src() {
+    return this._src;
+  }
+
+  set src(value) {
+    this.load(value).catch(console.error);
+  }
+
+  loadGLTF(src) {
+    return new Promise((resolve, reject) => {
+      new THREE.GLTFLoader().load(src, resolve, null, reject);
```

Comment:
creating new loader every time, not using `this` so it can be a function

Reply:
This method is overidden in ModelNode and we use the GLTFCache rather than the GLTFLoader there. There isn't much overhead to creating a new GLTFLoader and it doesn't need to be held onto after it is done loading. I actually think it might be better to create a loader per asset. There is some state that is associated with the currently loading model so you probably don't want to reuse it if you are loading in parallel.